canonicalwebteam/flask_base/logging.py

```diff
@@ -0,0 +1,190 @@
+    # https://github.com/nhairs/python-json-logger/blob/main/src/pythonjsonlogger/core.py
+    RESERVED_ATTRS: List[str] = [
+        "args",
+        "asctime",
+        "created",
+        "exc_info",
+        "exc_text",
+        "filename",
+        "funcName",
+        "levelname",
+        "levelno",
+        "lineno",
+        "module",
+        "msecs",
+        "message",
+        "msg",
+        "name",
+        "pathname",
+        "process",
+        "processName",
+        "relativeCreated",
+        "stack_info",
+        "thread",
+        "threadName",
+        "taskName",
+    ]
+
+    def __init__(self, fmt=None, datefmt=None):
+        super().__init__(fmt, datefmt)
+
+    def format(self, record: logging.LogRecord) -> str:
+        message = super().format(record)
+        extra_data = self._get_extra_dict(record)
+        if extra_data:
+            json_str = json.dumps(extra_data, indent=2, ensure_ascii=False)
+            return f"{message}\n{json_str}"
+        return message
+
+    def _get_extra_dict(self, record: logging.LogRecord) -> dict:
+        extra_dict = {}
+        for key, value in record.__dict__.items():
+            if (
+                key not in ExtraRichFormatter.RESERVED_ATTRS
+                and not key.startswith("_")
+            ):
+                extra_dict[key] = value
+
+        return extra_dict
+
+
+class GunicornDevLogger(GunicornLogger):
+    """
+    This logger use is optional and serves to display Gunicorn logs in the
+    same style as the default development application logs.
+    The way to use it is specified in the README.
+
+    MUST NOT BE USED in production.
+    """
+
+    def __init__(self, cfg):
+        super().__init__(cfg)
+
+    def setup(self, cfg):
+        super().setup(cfg)
+        self._substitute_stream_by_rich(self.error_log)
+        self._substitute_stream_by_rich(self.access_log)
+
+    def _substitute_stream_by_rich(self, logger):
+        for handler in list(logger.handlers):
+            if isinstance(handler, logging.StreamHandler):
+                logger.removeHandler(handler)
+                logger.addHandler(get_default_dev_handler())
+
+
+# Handlers (just one of each)
+
+
+@lru_cache(maxsize=1)
+def get_default_dev_handler() -> logging.Handler:
+    """
+    Handler to be used in development mode to get nice colored logs
+    """
+    rich_handler = RichHandler(
+        omit_repeated_times=False,
+        rich_tracebacks=True,
+        tracebacks_show_locals=True,
+        log_time_format=_date_format_with_ms,
+    )
+    rich_handler.setFormatter(
+        ExtraRichFormatter(
+            fmt=DEFAULT_DEV_FORMAT,
+        )
+    )
+    rich_handler.addFilter(RequestFilter())
+    return rich_handler
+
+
+@lru_cache(maxsize=1)
+def get_default_prod_handler() -> logging.Handler:
+    """
+    Handler to be used in production. Provides structured JSON logging.
+    """
+    log_handler = logging.StreamHandler()
+    formatter = JsonFormatter(
```

Comment:
I commented out the prod filter and was able to acheive the same results with 

```suggestion
    formatter = JsonFormatter(
        # the order of the format string doesn't matter
        # it just needs to include the fields that you want in the output
        # this is just for the default LogRecord attributes
        # for custom ones like "timestamp" check the RequestFilter class
        fmt="%(levelname)s:%(message)s",
        rename_fields={"levelname":"level"},
        timestamp=True
    )
```

but you still need a filter for the request id I'm guessing, also see https://nhairs.github.io/python-json-logger/latest/cookbook/#request-trace-ids

Reply:
Done